salt/modules/keystore.py

```diff
@@ -43,14 +41,14 @@ def _parse_cert(alias, public_cert, return_cert=False):
     ASN1 = OpenSSL.crypto.FILETYPE_ASN1
     PEM = OpenSSL.crypto.FILETYPE_PEM
     cert_data = {}
-    sha1 = public_cert.digest(b"sha1")
+    sha1 = public_cert.digest("sha1")
```

Comment:
Adding `.decode(__salt_system_encoding__)` to this line fixes the equality check in Python 3 comparing byte string against plain string https://github.com/saltstack/salt/blob/v3001.1/salt/states/keystore.py#L108.